### PR TITLE
feat: Add support for eglGetPlatformDisplayExt EGL Extension

### PR DESCRIPTION
**Pull request by caneraltinbasak**
#### Description of Change

Added eglGetPlatformDisplayExt EGL extension for project Thor. 

Thor EGL implementation includes all possible backends, Chromium should
use eglGetPlatformDisplayExt with EGL_PLATFORM_WAYLAND_KHR instead 
of eglGetDisplay to be able to get hold of the right backend.

### CODE REVIEW

File: patches/chromium/gl_add_support_for_eglgetplatformdisplayext_extension.patch [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
++}
++
+ __eglMustCastToProperFunctionPointerType GL_BINDING_CALL
+ MockEGLInterface::Mock_eglGetProcAddress(const char* procname) {
+   MakeEglMockFunctionUnique("eglGetProcAddress");
+@@ -799,10 +808,9 @@ static void MockEglInvalidFunction() {
+ 
+ GLFunctionPointerType GL_BINDING_CALL
+ MockEGLInterface::GetGLProcAddress(const char* name) {
+-  if (strcmp(name, "eglAcquireExternalContextANGLE") == 0) {
```
Comment (t-bashir-bs) [inline patches/chromium/gl_add_support_for_eglgetplatformdisplayext_extension.patch:33]: Was this change put in on purpose?

File: patches/chromium/gl_add_support_for_eglgetplatformdisplayext_extension.patch [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
++    return reinterpret_cast<GLFunctionPointerType>(
++        Mock_eglGetPlatformDisplayExt);
+   if (strcmp(name, "eglGetProcAddress") == 0)
+     return reinterpret_cast<GLFunctionPointerType>(Mock_eglGetProcAddress);
+   if (strcmp(name, "eglGetSyncAttrib") == 0)
+@@ -978,10 +989,9 @@ MockEGLInterface::GetGLProcAddress(const char* name) {
+   if (strcmp(name, "eglReacquireHighPowerGPUANGLE") == 0)
+     return reinterpret_cast<GLFunctionPointerType>(
+         Mock_eglReacquireHighPowerGPUANGLE);
+-  if (strcmp(name, "eglReleaseExternalContextANGLE") == 0) {
```
Comment (t-bashir-bs) [inline patches/chromium/gl_add_support_for_eglgetplatformdisplayext_extension.patch:55]: Was this change put in on purpose?

File: patches/chromium/gl_add_support_for_eglgetplatformdisplayext_extension.patch [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
++  if (strcmp(name, "eglReleaseExternalContextANGLE") == 0)
```
Comment (t-bashir-bs) [inline patches/chromium/gl_add_support_for_eglgetplatformdisplayext_extension.patch:56]: Was this change put in on purpose?

File: patches/chromium/gl_add_support_for_eglgetplatformdisplayext_extension.patch [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+ #define eglGetSyncAttribKHR ::gl::g_current_egl_context->eglGetSyncAttribKHRFn
+diff --git a/ui/gl/gl_bindings_autogen_mock.cc b/ui/gl/gl_bindings_autogen_mock.cc
+index 7ab5160ec4646d88a1993ee8b5be9966f6a957c5..6064e5d6c59532cfd6ba0b69ed3305a036971817 100644
+--- a/ui/gl/gl_bindings_autogen_mock.cc
++++ b/ui/gl/gl_bindings_autogen_mock.cc
+@@ -5708,9 +5708,8 @@ MockGLInterface::GetGLProcAddress(const char* name) {
+     return reinterpret_cast<GLFunctionPointerType>(Mock_glClientWaitSync);
+   if (strcmp(name, "glClientWaitSyncAPPLE") == 0)
+     return reinterpret_cast<GLFunctionPointerType>(Mock_glClientWaitSyncAPPLE);
+-  if (strcmp(name, "glClipControlEXT") == 0) {
```
Comment (t-bashir-bs) [inline patches/chromium/gl_add_support_for_eglgetplatformdisplayext_extension.patch:260]: Were these bracket changes put in on purpose?

**Review by t-bashir-bs — COMMENTED** [review]
Looks good, but do we need the bracket changes (even though I prefer having these)?

**Review by CFryerBrightSign — APPROVED** [review]

File: patches/chromium/gl_add_support_for_eglgetplatformdisplayext_extension.patch (hunk 1 above)
Comment (caneraltinbasak) [reply patches/chromium/gl_add_support_for_eglgetplatformdisplayext_extension.patch:33]: I've used the code autogenerator bundled in chromium itself:
ui/gl/generate_bindings.py

This generates stubs and bindings etc. for gl/egl calls, and at the end it runs clang-format.

This is due to clang-format version difference between chromium project and us. gclient updates itself, and clang-format. However, we are using Chromium120, which is not the latest. So when we run latest clang-format with older code, it makes changes. It is too much hassle to go back to depot-tools version deployed at the time when Chromium120 is released.

**Review by t-bashir-bs — APPROVED** [review]

**Pull request merged**